Core/Foundation/Include/Babylon/DebugTrace.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <functional>
+
+#define DEBUG_TRACE_NOOP(...)
+
+#define _STRINGIFY(x) #x
+#define _TOSTRING(x) _STRINGIFY(x)
+#define _DEBUG_TRACE(_format, ...) Babylon::DebugTrace::Trace(__FILE__ "(" _TOSTRING(__LINE__) "): " _format "\n", ##__VA_ARGS__); 
+
+#if JSRUNTIMEHOST_DEBUG_TRACE
+#   define DEBUG_TRACE _DEBUG_TRACE
+#else
+#   define DEBUG_TRACE(...) DEBUG_TRACE_NOOP()
+#endif // JSRUNTIMEHOST_DEBUG_TRACE
+
+namespace Babylon
+{
+    namespace DebugTrace
+    {
+        // off by default.
+        void SetDebugTrace(bool enabled);
```

Comment:
Maybe a better name?
```suggestion
        void EnableDebugTrace(bool enabled);
```